zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -514,4 +516,87 @@ mod test {
             test_ok(mock_tx(value, gas_price, calldata), code);
         }
     }
+
+    #[test]
+    fn begin_tx_no_code() {
+        let block: GethData = TestContext::<2, 1>::new(
+            None,
+            |accs| {
+                accs[0].address(MOCK_ACCOUNTS[0]).balance(eth(20));
+                accs[1].address(MOCK_ACCOUNTS[1]).balance(eth(10));
+            },
+            |mut txs, _accs| {
+                txs[0]
+                    .from(MOCK_ACCOUNTS[0])
+                    .to(MOCK_ACCOUNTS[1])
+                    .gas_price(gwei(2))
+                    .gas(Word::from(0x10000))
+                    .value(eth(2));
+            },
+            |block, _tx| block.number(0xcafeu64),
+        )
+        .unwrap()
+        .into();
+
+        assert_eq!(run_test_circuit_geth_data_default::<Fr>(block), Ok(()));
+    }
+
+    #[test]
+    fn begin_tx_no_account() {
+        let block: GethData = TestContext::<1, 1>::new(
+            None,
+            |accs| {
+                accs[0].address(MOCK_ACCOUNTS[0]).balance(eth(20));
+            },
+            |mut txs, _accs| {
+                txs[0]
+                    .from(MOCK_ACCOUNTS[0])
+                    .to(MOCK_ACCOUNTS[1])
+                    .gas_price(gwei(2))
+                    .gas(Word::from(0x10000))
+                    .value(eth(2));
+            },
+            |block, _tx| block.number(0xcafeu64),
+        )
+        .unwrap()
+        .into();
+
+        assert_eq!(run_test_circuit_geth_data_default::<Fr>(block), Ok(()));
+    }
+
+    // TODO: Enable this test once we have support for contract deployment from
+    // BeginTx.
+    #[ignore]
```

Comment:
I also noticed that this test passes, but I think that's a coincidence.  I disabled it because currently we don't support this feature, and I was afraid that some change in a future PR (before supporting contract deployment) may make this test fail.

The reason why it passes now I think is because the circuits are treating the destination account as an account with no code, so they verifies an ETH transfer without code execution.

Reply:
Sounds good. Thanks for explaining this.